### ISSUE TITLE
How to config "FFmpeg:probesize" and “FFmpeg:analyzeduration”

### ISSUE THREAD

**Issue by ssyangchn** [jellyfin/jellyfin#3930]
**Describe the bug**

1. Im trying config the "FFmpeg:probesize" and “FFmpeg:analyzeduration” but errors in the log
```
.\jellyfin.exe --datadir "d:\ProgramData\Jellyfin\Server" --FFmpeg:probesize "50M" --FFmpeg:analyzeduration "10M"
Jellyfin.Server 10.6.3.0
Copyright ?  2019 Jellyfin Contributors. Code released under the GNU General Public License

ERROR(S):
  Option 'FFmpeg:probesize' is unknown.
  Option 'FFmpeg:analyzeduration' is unknown.
```
2. How to config jellyfin use a proxy ? not a reverse proxy

The [jellyfin doc: ](https://jellyfin.org/docs/general/administration/configuration.html#main-configuration-options)

```
Main Configuration
The main server configuration is built upon the ASP .NET configuration framework, which provides a tiered approach to loading configuration. The base directory to locate the configuration files is set using the configuration directory setting. The configuration sources are as follows, with later sources having higher priority and overwriting the values in earlier sources.

Hard-coded default values: These defaults are specified specified in the Jellyfin source code and cannot be changed.
Default logging configuration file (logging.default.json): This file should not be modified manually by users. It is reserved by the server to be overwritten with new settings on each new release.
System-specific logging configuration file (logging.json): This is the file you should change if you want to have a custom logging setup. Jellyfin uses the Serilog logging framework, and you can read about the configuration options available in their documentation.

NOTE
This file can be changed at runtime, which will automatically reload the configuration and apply the changes immediately.

Environment variables: The documentation provided by Microsoft explains how to set these configuration options via environment variables. Jellyfin uses its own custom JELLYFIN_ prefix for these variables. For example, to set a value for the HttpListenerHost:DefaultRedirectPath setting, you would set a value for the JELLYFIN_HttpListenerHost__DefaultRedirectPath environment variable.

Command line options: Certain command line options are loaded into the configuration system and have the highest priority. The following command line options are mapped to associated configuration options.

--nowebcontent sets the hostwebclient configuration setting to false
--plugin-manifest-url sets a value for the InstallationManager:PluginManifestUrl configuration setting
Main Configuration Options
This section lists all the configuration options available and explains their function.
```
Key | Default Value | Description
-- | -- | --
hostwebclient | True | Set to True if the server should host the web client.
HttpListenerHost:DefaultRedirectPath | "web/index.html" if hostwebclient is true; "swagger/index.html" if hostwebclient is false | The default redirect path to use for requests where the URL base prefix is invalid or missing
InstallationManager:PluginManifestUrl | "https://repo.jellyfin.org/releases/plugin/manifest.json" | The URL for the plugin repository JSON manifest.
FFmpeg:probesize | "1G" | Value to set for the FFmpeg probesize format option. See the FFmpg documentation for more details.
FFmpeg:analyzeduration | "200M" | The value to set for the FFmpeg analyzeduration format option. See the FFmpg documentation for more details.
playlists:allowDuplicates | True | Whether playlists should allow duplicate items or automatically filter out duplicates.
PublishedServerUrl | Server Url based on primary IP address | The Server URL to publish in udp Auto Discovery response.

**System (please complete the following information):**
 - OS: windows
 - Jellyfin Version:10.6.3

**Comment by Bond-009** [comment]
1. Set en environment variable JELLYFIN_FFmpeg__analyzeduration

**Comment by chr00189** [comment]
Hi - trying to pass values to probesize and analyizeduration on a Buster linux install. How are these passed in linux? There does not appear to be environment variables for these in linux.

**Comment by thulle** [comment]
In case anyone passes by with this question, ENV variables goes into /etc/default/jellyfin

**Comment by jrbski** [comment]
In addition to the important hint that ENV variables goes into /etc/default/jellyfin on linux, the other detail from the microsoft asp.net doc related to linux is:

**"The : separator doesn't work with environment variable hierarchical keys on all platforms. __, the double underscore, is
supported by all platforms. For example, the : separator is not supported by [Bash](https://linuxhint.com/bash-environment-variables/), but __ is."**

That is why the variable "FFmpeg:analyzeduration" needs to be "FFmpeg__analyzeduration" in linux

**Comment by DumberSoft** [comment]
@jrbski @Bond-009 I have a query how can I change the environment JELLYFIN_FFmpeg__analyzeduration with 30M o 3000000 🤔

**Comment by jrbski** [comment]
> 

Reducing this value definitely reduces the liveTV startup time to 2 seconds or so for me, vs. 10+ seconds with the 200M default. The 'old' default of 3000000 (or 3M) does not seem to cause any failures to start transcoding  for me.  I'm using an Atomic PI as my jellyfin server so YMMV (but since I'm successfully using a Atom x5-Z8350, I suspect that faster CPU's may work even better!). 

To change it on Linux, edit  /etc/default/jellyfin, and add a new line to the file with "JELLYFIN_FFmpeg__analyzeduration= 3000000", and restart Jellyfin to make it effective. (I'm not sure where the equivalent config file is in Windows.) In my case I added it into the config file here:

 `# Space to add additional command line options to jellyfin (for help see ~$ jellyfin --help)
JELLYFIN_ADDITIONAL_OPTS=""
**JELLYFIN_FFmpeg__analyzeduration=3000000**

#[OPTIONAL] run Jellyfin with ASP.NET Server Garbage Collection (uses more RAM and less CPU than Workstation GC)
#0 = Workstation
#1 = Server
#COMPlus_gcServer=1
`

You can check the transcode logs to confirm that the new value is being used by looking for a line similar to this:
/usr/lib/jellyfin-ffmpeg/ffmpeg **-analyzeduration 3000000** -fflags +igndts -f mpegts -autorotate 0 -i "http://192.168.###.###:8096/LiveTv/LiveStreamFiles/e230300fd49148f1bb930381c8b128ae/stream.ts" -map_metadata -1 -map_chapters -1 -threads 0 -sn -codec:v:0 libx264 -preset superfast -crf 25 -maxrate 1616000 -bufsize 3232000 -profile:v:0 high -level 51 -x264opts:0

**Comment by DumberSoft** [comment]
In my case I am using a paid VPS server, can it also be there, since it is 60 GB of RAM 10 Cores but it is AMD EPYC?

**Comment by jrbski** [comment]
> In my case I am using a paid VPS server, can it also be there, since it is 60 GB of RAM 10 Cores but it is AMD EPYC?

Not sure what you are using for your liveTV source and how you have it set-up. I'm using an HDHomerun Prime, but it is on a local gigabit network. Getting a tunnel back to your tuner from the VPS server could be interesting, and any latency between the VPS server and your tuner could be part of your challenge. You definitely have plenty of server horsepower. My atomic PI has only 2 GB of ram..

**Comment by DumberSoft** [comment]
Well I am using IPTV servers but I am sending it to m3u but that is my answer that has intensive freezes at the time of putting it, for that reason I would like to know if it can be solved with what you are indicating about JELLYFIN_FFmpeg__analyzeduration.

**Comment by YOYOMAMA** [comment]
Thanks @jrbski for the detailed writeup above!
Trying to do the exact same thing but its just not working for me on jellyfin_10.8.4 / ubuntu.

I created and added the following line to /etc/default/jellyfin and restarted jellyfin (per thread above):
JELLYFIN_FFmpeg__analyzeduration=3000000
(/etc/default/jellyfin has full permissions.. tried 644 and 777.)

Testing with tuning into a live tv station and I unfortunately still see evidence in /opt/jellyfin/log/log_20221207.log that analyzeduration is still 200M (eg my changes to the env var is not taking effect :(  ): 
[2022-12-07 18:57:06.777 -06:00] [INF] [58] Jellyfin.Api.Helpers.TranscodingJobHelper: "/usr/bin/ffmpeg" "-analyzeduration **200M** -fflags +igndts -f mpegts -autorotate 0 -i \"http:// .......

(.. as a result, live tv is still taking a long time to load.  Hoping this setting will help.)

Lots of talk about this being the right way to set this variable and I've tried for several days but its just not working and still scratching my head.

Any insights/help would be much appreciated for what I'm sure will turn out to be a silly question.
